src/util/__tests__/createServerRootMixin.test.js

```diff
@@ -576,33 +623,37 @@ Array [
             indexName: 'movies',
           }),
         ],
-        render(h) {
-          return h(InstantSearchSsr, {}, [
-            h(Configure, {
-              attrs: {
-                hitsPerPage: 100,
-              },
-            }),
+        render: renderCompat(h =>
+          h(InstantSearchSsr, {}, [
+            h(
+              Configure,
+              isVue3
+                ? { hitsPerPage: 100 }
+                : {
+                    attrs: {
+                      hitsPerPage: 100,
+                    },
+                  }
+            ),
             h(SearchBox),
-          ]);
-        },
-        // in test, beforeCreated doesn't have $data yet, but IRL it does
+          ])
+        ),
         created() {
           this.instantsearch.hydrate({
             movies: nonSerialized,
           });
-        },
-      };
 
-      const {
-        vm: { instantsearch },
-      } = mount(app);
+          expect(this.instantsearch.__initialSearchResults).toEqual(
+            expect.objectContaining({ movies: expect.any(SearchResults) })
+          );
 
-      expect(instantsearch.__initialSearchResults).toEqual(
-        expect.objectContaining({ movies: expect.any(SearchResults) })
-      );
+          expect(this.instantsearch.__initialSearchResults.movies).toEqual(
```

Comment:
I'm not sure why the object reference is different, but it might have something to do with mocking or something around it. Not sure, though.

Reply:
is it with the serialising?

Reply:
ah right. I got confused with other test cases where it used to be `toBe` but I had to replace with `toEqual`. In this case it makes sense to use `toEqual`.